cmd/rpcdaemon/commands/rpc_block.go

```diff
@@ -43,3 +64,37 @@ func getLatestBlockNumber(tx kv.Tx) (uint64, error) {
 
 	return blockNum, nil
 }
+
+func getFinalizedBlockNumber(tx kv.Tx) (uint64, error) {
+	forkchoiceFinalizedHash := rawdb.ReadForkchoiceFinalized(tx)
+	if forkchoiceFinalizedHash != (common.Hash{}) {
+		forkchoiceFinalizedNum := rawdb.ReadHeaderNumber(tx, forkchoiceFinalizedHash)
+		if forkchoiceFinalizedNum != nil {
+			return *forkchoiceFinalizedNum, nil
+		}
+	}
+
+	blockNum, err := stages.GetStageProgress(tx, stages.Execution)
+	if err != nil {
+		return 0, fmt.Errorf("getting latest finalized block number: %w", err)
+	}
+
+	// finalized block is the genesis block
+	if blockNum-128 <= 0 {
+		return 0, nil
+	}
+
+	return blockNum - 128, nil
```

Comment:
Is finalised block always -128 from execution progress?

Reply:
"Refer to a block that is likely to stay in the PoW canonical chain forever, i.e. a block which is e.g. 128 blocks behind the head" I just followed this, but I am unsure now wether the respond pre-merge should be an error or 128 blocks behind.

Reply:
Geth is returning nil, so I have changed our response to be 0